cpp/include/raft/spatial/knn/detail/topk/warpsort_topk.cuh

```diff
@@ -591,9 +592,10 @@ class block_sort {
   }
 
   /** Save the content by the pointer location. */
-  __device__ void store(T* out, IdxT* out_idx) const
+  template <typename Lambda = raft::identity_op>
+  __device__ void store(T* out, IdxT* out_idx, Lambda post_process = raft::identity_op()) const
```

Comment:
Just for the record: I'm currently refactoring select_k in https://github.com/rapidsai/raft/pull/1085 , so I'd need to catch up with these changes there. I'm also planning to change `T` and `IdxT` into `OutT` and `OutIdxT` to allow saving the results in a type different from the working types `T` and `IdxT`.